### PR TITLE
Refactor tests to use page objects

### PR DESCRIPTION
**Pull request by eablack**
https://github.com/code-corps/code-corps-ember/issues/381

### CODE REVIEW

**Comment by WenInCode** [comment]
@eablack Any chance you can create an issue for this so it's clear that it's in progress for others

See: https://docs.google.com/document/d/1KpppRH8lBx4G_buFbgeW1UbfLTSf8jqqFq08Kq47svw/edit

Thanks!

**Comment by noslouch** [comment]
There's a lot of UI interaction in here that should encapsulated by a page object. All those clicks and fillIns should be part of a page object's api.

File: tests/acceptance/footer-test.js [hunk 1]
```diff
@@ -1,10 +1,11 @@
...
+import indexPage from '../pages/index';
 
 moduleForAcceptance('Acceptance | footer');
 
 test('can visit links from footer', function(assert) {
-  visit('/');
+  indexPage.visit();
 
   andThen(function() {
     click('.site-footer ul > li:eq(1) li:eq(0) a');
```
Comment (noslouch) [inline tests/acceptance/footer-test.js]: For these kinds of "global" UI elements, you can inline a page object component on the index page object that's scoped to the site-footer, but what would be even better would be to add a new site footer component file in tests/pages/components/site-footer.js that other page objects can import.

File: tests/pages/organization-index.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import {
+  create,
+  visitable
+} from 'ember-cli-page-object';
+
+export default create({
+  visit: visitable(':organization')
+});
```
Comment (noslouch) [inline tests/pages/organization-index.js]: These and the other page objects are a good start, but there's much more cool stuff you can do with page objects beyond setting up a visit call. All the UIcan be encapsulated in here as well. Take a look at https://github.com/code-corps/code-corps-ember/blob/develop/tests/pages/onboarding.js for some samples.

**Comment by eablack** [comment]
Cool, @noslouch. Updated to utilize page object UI interactions. Also added that site footer page object component. Have another look?

File: tests/acceptance/organization-test.js [hunk 3]
```diff
@@ -78,19 +79,19 @@ test('an admin can navigate to settings', (assert) => {
...
-  visit(organization.slug);
+  organizationPage.visitIndex({ organization: organization.slug });
 
   andThen(() => {
     assert.ok(find('.organization-menu li a:contains("Projects")').hasClass('active'), 'The organization projects menu is active');
     Ember.run.next(() => {
-      click('.organization-menu li a:contains("Settings")');
+      organizationPage.clickSettings();
+      andThen(() => {
+        assert.ok(find('.organization-menu li a:contains("Settings")').hasClass('active'), 'The organization settings menu is active');
```
Comment (WenInCode) [inline tests/acceptance/organization-test.js]: Can this be added to a page object?

File: tests/acceptance/organization-test.js [hunk 4]
```diff
@@ -78,19 +79,19 @@ test('an admin can navigate to settings', (assert) => {
   // showing/hiding of links is handled in the organization-menu component integration test
   authenticateSession(application, { user_id: user.id });
 
-  visit(organization.slug);
+  organizationPage.visitIndex({ organization: organization.slug });
 
   andThen(() => {
     assert.ok(find('.organization-menu li a:contains("Projects")').hasClass('active'), 'The organization projects menu is active');
```
Comment (WenInCode) [inline tests/acceptance/organization-test.js]: Can this be added to a page object

**Comment by WenInCode** [comment]
A lot of good work here - just updated with a 2 questions

**Comment by eablack** [comment]
Thanks, @WenInCode. Updated to use the page object for checking the class of the active menu item.

**Comment by WenInCode** [comment]
This looks good to me can you squash the commits (ref: https://github.com/code-corps/code-corps-ember/blob/develop/docs/SQUASHING.md) and merge this (if you have access for merging, otherwise just let me know). Thanks for all your hard work @eablack

**Comment by eablack** [comment]
@WenInCode cool, updated the fork and squashed to one commit. Looks like I don't have write access, so could you merge for me?

**Comment by eablack** [comment]
Nevermind, I have write access after all!

**Pull request merged**